pkgs/development/libraries/gts/default.nix

```diff
@@ -1,34 +1,65 @@
-{ fetchurl, stdenv, pkgconfig, autoreconfHook, gettext, glib }:
+{ fetchurl, stdenv, pkgconfig, autoreconfHook, gettext, glib, buildPackages }:
 
+# gts generates binaries during build that will be used during the build process.
+# In case of cross-platform  builds, this package needs to bootstrap itself
+# in order to generate these binaries for the propper build architecture.
+# Please keep that in mind when making changes to the derivation.
 
-stdenv.mkDerivation rec {
-  pname = "gts";
-  version = "0.7.6";
+# When making significant changes, please try to test cross-platform builds as well.
+# Example:
+#   nix-build . -A pkgsCross.aarch64-multiplatform.gts
+#   nix-shell -p qemu --run "qemu-aarch64 result-bin/bin/transform --help"
 
-  outputs = [ "bin" "dev" "out" ];
+let
 
-  src = fetchurl {
-    url = "mirror://sourceforge/gts/${pname}-${version}.tar.gz";
-    sha256 = "07mqx09jxh8cv9753y2d2jsv7wp8vjmrd7zcfpbrddz3wc9kx705";
-  };
+  cross = stdenv.hostPlatform != stdenv.buildPlatform;
 
-  nativeBuildInputs = [ pkgconfig autoreconfHook ];
-  buildInputs = [ gettext ];
-  propagatedBuildInputs = [ glib ];
+  name = "gts";
+  drv = { pname ? name } : rec {
+    inherit pname;
+    version = "0.7.6";
 
-  doCheck = false; # fails with "permission denied"
+    outputs = [ "bin" "dev" "out" ];
 
-  meta = {
-    homepage = "http://gts.sourceforge.net/";
-    license = stdenv.lib.licenses.lgpl2Plus;
-    description = "GNU Triangulated Surface Library";
+    depsBuildBuild = [ buildPackages.stdenv.cc ];
 
-    longDescription = ''
-      Library intended to provide a set of useful functions to deal with
-      3D surfaces meshed with interconnected triangles.
-    '';
+    src = fetchurl {
+      url = "mirror://sourceforge/gts/${name}-${version}.tar.gz";
+      sha256 = "07mqx09jxh8cv9753y2d2jsv7wp8vjmrd7zcfpbrddz3wc9kx705";
+    };
+
+    # glib needs to be in nativeBuildInputs in order for AM_PATH_GLIB_2_0
```

Comment:
This is suspicious, I would expect  the macros to be picked up just like pkg-config files are.

Reply:
The macro itself is not the issue, it's the AC_TRY_RUN that is used by AM_PATH_GLIB_2_0 internally which will fail if not run in native context.